### PR TITLE
Abstract common behavior of test classes.

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
Add abstraction for test classes. Abstract check for running workspace as is used in almost all test classes as @Before method. Abstract method for opening file.

Tested localy against prod-preview.

### CODE REVIEW

**Comment by Katka92** [comment]
[test]

File: functional-tests/src/test/java/com/redhat/che/functional/tests/RhCheAbstractTestClass.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      LOG.error(
+          "Could not obtain workspace name and id - worskape was probably not successfully injected.");
+      throw e;
+    } catch (Exception e) {
+      LOG.error("Could not open workspace IDE.");
+      throw e;
+    }
+  }
+
+  public void openDefinedClass(String projectFile) {
```
Comment (rhopp) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/RhCheAbstractTestClass.java]: Couldn't this method be used instead? https://github.com/eclipse/che/blob/bb8e6161b53616aedefad18c066998d7cd0e5c5f/selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java#L1092

Comment (Katka92) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/RhCheAbstractTestClass.java]: I've removed this method and used upstream one. I decided to use this method: 
https://github.com/eclipse/che/blob/bb8e6161b53616aedefad18c066998d7cd0e5c5f/selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java#L1052
Although it requires more code added, it seems more stable when testing localy.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  protected void removeDependency() {
+    editor.setCursorToLine(INJECTION_ENTRY_POINT);
+    editor.deleteCurrentLine();
+    editor.waitTabFileWithSavedStatus(PROJECT_FILE);
+  }
+
+  protected void editorCheckBayesianError() {
+    if (testApiEndpointUrlProvider.get().getHost().equals(CHE_PROD_PREVIEW_URL)) {
```
Comment (rhopp) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianAbstractTestClass.java]: I don't think this is the right approach... In this case, when bayesian is fixed in prod-preview, this test will be still skipped...
IMHO we want to skip the test only if it fails AND the env is prod-preview.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java [hunk 3]
```diff
@@ -11,139 +11,38 @@
...
+    super.setExpectedErrorLine(JSON_EXPECTED_ERROR_LINE);
+    super.setExpectedErrorText(JSON_EXPECTED_ERROR_TEXT);
+    super.setInjectionEntryPoint(JSON_INJECTION_ENTRY_POINT);
+    super.setProjectFile(PROJECT_FILE);
+    super.setPathToFile(PATH_TO_FILE);
+    super.setErrorMessage(ERROR_MESSAGE);
+    super.setProjectName(PROJECT_NAME);
+    super.setProjectDependency(PROJECT_DEPENDENCY);
+    super.setWorksapce(workspace);
+    super.openTestFile();
```
Comment (rhopp) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java]: I don't think these `super` are necessary...

File: functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java [hunk 4]
```diff
@@ -57,102 +39,30 @@
...
+    super.setExpectedErrorLine(POM_EXPECTED_ERROR_LINE);
+    super.setExpectedErrorText(POM_EXPECTED_ERROR_TEXT);
+    super.setInjectionEntryPoint(POM_INJECTION_ENTRY_POINT);
+    super.setProjectFile(PROJECT_FILE);
+    super.setPathToFile(PATH_TO_FILE);
+    super.setErrorMessage(ERROR_MESSAGE);
+    super.setProjectName(PROJECT_NAME);
+    super.setProjectDependency(PROJECT_DEPENDENCY);
+    super.setWorksapce(workspace);
+    super.openTestFile();
```
Comment (rhopp) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPomXml.java]: Same thing.

**Pull request merged**